### DISCUSSION TITLE
Ability to fallback to the native RegExp OR be able to change regex engine

### DISCUSSION THREAD

**Discussion by songkeys** [renovatebot/renovate#8040] [Request Help] [answered]
**What would you like Renovate to be able to do?**

I'm trying to specify a regex for `ignoreDeps` to allow only a certain group of packages to be checked and updated, but it turns out that `ignoreDeps` only supports strings.

So I turned to the `packageRules` with the following configuration:

```js
    packageRules: [
      {
        packagePatterns: ["^(?!@abc)"], // i.e. i only want packages like `@abc/xxx` to be checked
        enabled: false,
      },
    ],
```

As you can see here, I used the negative lookahead feature of regex, which is unfortunately [not supported by `re2` engine](https://github.com/uhop/node-re2#limitations-things-re2-does-not-support). It will throw a `SyntaxError`.

**Did you already have any implementation ideas?**

Here's some suggestions and solutions:

1. The re2 library suggests us using a try catch fallback for the missing features, which seems quite reasonable for renovate.
2. Or, there should be an option to let us choose which regex engine to use. (I know there are some potential vulnerabilities but the re2 library is not perfect under some cases.)
3. Add `allowDeps` option with regex patterns so that I could use `^@abc` instead. (Some other options like `ignoreDeps` and `ignorePaths` should also be supported with regex patterns for consistence.)

**Comment by viceice** [comment, marked as answer]
You can simply use [excludePackagePatterns](https://docs.renovatebot.com/configuration-options/#excludepackagepatterns)

```js
    packageRules: [
      {
        excludePackagePatterns: ["^@abc\\/"], // i.e. i only want packages like `@abc/xxx` to be checked
        enabled: false,
      },
    ],
```

or

**Reply by songkeys** [reply]
Thank you so much. This configuration works! I missed this option.

Adding this suggestion to <https://docs.renovatebot.com/configuration-options/#ignoredeps> section might help.

**Comment by rarkins** [comment]
FYI the reason for using re2 is to prevent DDoS attacks, so we're not going to fall back to regular regex for that reason.
